cpp/src/arrow/type.h

```diff
@@ -67,9 +67,12 @@ struct Type {
     // Variable-length bytes (no guarantee of UTF8-ness)
     BINARY,
 
-    // By default, int32 days since the UNIX epoch
+    // int64_t milliseconds since the UNIX epoch
     DATE,
 
+    // int32_t days since the UNIX epoch
+    DATE32,
```

Comment:
I don't think so, and I don't think we should guarantee ABI stability of the enum values yet anyhow (at some point we will want to)